src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs

```diff
@@ -148,15 +148,21 @@ private static HttpConnectionKey GetConnectionKey(HttpRequestMessage request, Ur
         {
             Uri uri = request.RequestUri;
 
+            // TODO: #28863 Part 1: Uri.IdnHost should include [] around IPv6 address.
             // If the hostname is an IPv6 address, uri.IdnHost will return the address without enclosing [].
             // In this case, use uri.Host instead, which will correctly enclose with [].
             // Note we don't need punycode encoding if it's an IP address, so using uri.Host is fine.
-            bool isIPv6Address = uri.HostNameType == UriHostNameType.IPv6;
+
+            // TODO: #28863 Part 2: Uri.Host LLA (Link-local address) for IPv6 address doesn't contain %number part.
+            // Since scope is mandatory for LLA, we will use uri.IdnHost for IPv6 LLA address.
+            // From RFC 4291, IPv6 LLA has formate like: fe80::/10.
+            bool isNonLLAIPv6Address = (uri.HostNameType == UriHostNameType.IPv6 &&
+                uri.IdnHost.IndexOf("fe80") != 0);
```

Comment:
Alternatively, we can use `IPAddress.IsIPv6LinkLocal` Property, but that requirs [convert uri to IPAddress first](https://github.com/dotnet/corefx/blob/master/src/System.Net.Primitives/src/System/Net/IPAddressParser.cs#L27), which is an expensive operation.

The logic here to determine `isNonLLAIPv6Address` is based on [`IPAddress.IsIPv6LinkLocal`.](https://github.com/dotnet/corefx/blob/008c119440aa5219cc3c274a3a1edfa8c21711f3/src/System.Net.Primitives/src/System/Net/IPAddress.cs#L467)

Reply:
If anything it must start with the prefix: fe80::/10 The IndexOf will try to find it anywhere in the string, right? 
The documentation says uri.IdnHost is string suitable for DNS lookup. 
It should not include [] IMHO but the constructed URL should e.g. "[fe80::1%22]"

Reply:
> The IndexOf will try to find it anywhere in the string, right?

Yes, but here is intentionally checked for the index != 0 case, to make sure that it's not IPv6 LLA.

> It should not include [] IMHO but the constructed URL should e.g. "[fe80::1%22]"

I think so. Will make URL to contain `[]` and add more test cases in next commit.